shared_libs/event_logger_lib/event_logger_lib/event_logger.py

```diff
@@ -0,0 +1,123 @@
+import logging
+import os
+from dataclasses import asdict
+from datetime import datetime, timezone
+from typing import Optional
+
+from azure.identity import DefaultAzureCredential, ManagedIdentityCredential
+from azure.monitor.opentelemetry import configure_azure_monitor
+
+from .log_event import LogEvent, EventType
+
+logger = logging.getLogger(__name__)
+
+
+class EventLogger:
+
+    def __init__(self, workflow_id: str, microservice_id: str):
+        self.workflow_id = workflow_id
+        self.microservice_id = microservice_id
+        connection_string = os.getenv("APPLICATIONINSIGHTS_CONNECTION_STRING", "").strip()
+        self.azure_monitor_enabled = bool(connection_string)
+
+        if self.azure_monitor_enabled:
+            self._initialize_azure_monitor()
+        else:
+            logger.info("Azure Monitor logging is disabled - APPLICATIONINSIGHTS_CONNECTION_STRING not set or empty. "
+                        "Standard logger will be used.")
+
+    def _initialize_azure_monitor(self) -> None:
+        try:
+            credential = self._get_credential()
+            configure_azure_monitor(
+                credential=credential,
+                logger_name=__name__,
+            )
+            logger.info("Azure Monitor initialized successfully")
+        except Exception as e:
+            logger.error(f"Failed to initialize Azure Monitor: {e}")
+            self.azure_monitor_enabled = False
+            raise
```

Comment:
Would raising this exception halt all processing if monitoring cannot be initialised and do we want that to happen say on prod?

Reply:
Okay sound